Hi @thiagokokada !
I just ran a nixpkgs-review and got two errors. One in nix build and the other when running the tool itself. The error in nix build is found when compiling crackmapexec.

```
error: builder for '/nix/store/j72ppqdkbaa7ny5zvln69yv9vbvi06n9-crackmapexec-5.4.0.drv' failed with exit code 1;
       last 10 log lines:
       > Finished creating a wheel...
       > /build/source/dist /build/source
       > Unpacking to: unpacked/crackmapexec-5.4.0...OK
       > Repacking wheel as ./crackmapexec-5.4.0-py3-none-any.whl...OK
       > /build/source
       > Finished executing pypaBuildPhase
       > Running phase: pythonRuntimeDepsCheckHook
       > Executing pythonRuntimeDepsCheck
       > Checking runtime dependencies for crackmapexec-5.4.0-py3-none-any.whl
       >   - neo4j not satisfied by version 5.16.dev0
       For full logs, run 'nix log /nix/store/j72ppqdkbaa7ny5zvln69yv9vbvi06n9-crackmapexec-5.4.0.drv'.
error: 1 dependencies of derivation '/nix/store/7b06w3ldjcnqny98v6zbzfmwlr3pzf4q-review-shell.drv' failed to build
```
It seems to be a similar dependency error as dploot or pywerview, this time with neo4j.

The other error is when running nxc with smb or ssh does not find the module impacket.dcerpc.v5.gkdi:

```
Traceback (most recent call last):
  File "/nix/store/66fi4py61wqr9zxhbvb94qjfmb7q6caw-netexec-1.1.0-unstable-2024-01-15/bin/.nxc-wrapped", line 9, in <module>
    sys.exit(main())
             ^^^^^^
  File "/nix/store/66fi4py61wqr9zxhbvb94qjfmb7q6caw-netexec-1.1.0-unstable-2024-01-15/lib/python3.11/site-packages/nxc/netexec.py", line 152, in main
    protocol_object = getattr(p_loader.load_protocol(protocol_path), args.protocol)
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/66fi4py61wqr9zxhbvb94qjfmb7q6caw-netexec-1.1.0-unstable-2024-01-15/lib/python3.11/site-packages/nxc/loaders/protocolloader.py", line 16, in load_protocol
    loader.exec_module(protocol)
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/nix/store/66fi4py61wqr9zxhbvb94qjfmb7q6caw-netexec-1.1.0-unstable-2024-01-15/lib/python3.11/site-packages/nxc/protocols/ssh.py", line 9, in <module>
    from nxc.connection import connection, highlight
  File "/nix/store/66fi4py61wqr9zxhbvb94qjfmb7q6caw-netexec-1.1.0-unstable-2024-01-15/lib/python3.11/site-packages/nxc/connection.py", line 13, in <module>
    from nxc.protocols.ldap.laps import laps_search
  File "/nix/store/66fi4py61wqr9zxhbvb94qjfmb7q6caw-netexec-1.1.0-unstable-2024-01-15/lib/python3.11/site-packages/nxc/protocols/ldap/laps.py", line 11, in <module>
    from impacket.dcerpc.v5.gkdi import MSRPC_UUID_GKDI, GkdiGetKey, GroupKeyEnvelope
ModuleNotFoundError: No module named 'impacket.dcerpc.v5.gkdi'
```

In this other case, the truth is that I don't know why it doesn't detect the module if it is present in impacket.

--- a/pkgs/development/python-modules/dploot/default.nix
+++ b/pkgs/development/python-modules/dploot/default.nix
@@ -1,0 +1,48 @@
+{ lib
+, buildPythonPackage
+, poetry-core
+, pythonRelaxDepsHook
+, fetchPypi
+, impacket
+, cryptography
+, pyasn1
+, lxml
+}:
+
+buildPythonPackage rec {
+  pname = "dploot";
+  version = "2.2.4";
+  pyproject = true;
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-40/5KOlEFvPL9ohCfR3kqoikpKFfJO22MToq3GhamKM=";
+  };
+
+  pythonRelaxDeps = true;
+
+  nativeBuildInputs = [
+    poetry-core
+    pythonRelaxDepsHook
+  ];
+
+  propagatedBuildInputs = [
+    impacket
+    cryptography
+    pyasn1
+    lxml
+  ];
+
+  pythonImportsCheck = [ "dploot" ];
+
+  # No tests
+  doCheck = false;
+
+  meta = {
+    homepage = "https://github.com/zblurx/dploot";
+    description = "DPAPI looting remotely in Python";
+    changelog = "https://github.com/zblurx/dploot/releases/tag/${version}";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ vncsb ];
+  };
+}

--- a/pkgs/tools/security/netexec/default.nix
+++ b/pkgs/tools/security/netexec/default.nix
@@ -58,25 +58,6 @@ let
     };
   };
 
-  dploot = buildPythonPackage rec {
-    pname = "dploot";
-    version = "2.2.4";
-    pyproject = true;
-    src = fetchPypi {
-      inherit pname version;
-      hash = "sha256-40/5KOlEFvPL9ohCfR3kqoikpKFfJO22MToq3GhamKM=";
-    };
-    nativeBuildInputs = with python.pkgs; [
-      poetry-core
-    ];
-    propagatedBuildInputs = with python.pkgs; [
-      impacket
-      cryptography
-      pyasn1
-      lxml
-    ];
-  };
-
   resource = buildPythonPackage rec {
     pname = "resource";
     version = "0.2.1";

--- a/pkgs/tools/security/netexec/default.nix
+++ b/pkgs/tools/security/netexec/default.nix
@@ -16,11 +16,6 @@ let
           hash = "sha256-Jozn4lKAnLQ2I53+bx0mFY++OH5P4KyqVmrS5XJUY3E=";
         };
       };
-      bloodhound-py = super.bloodhound-py.overridePythonAttrs (old: {
-        propagatedBuildInputs =
-          lib.lists.remove super.impacket old.propagatedBuildInputs
-          ++ [ self.impacket ];
-      });
     };
   };
 

--- a/pkgs/tools/security/netexec/default.nix
+++ b/pkgs/tools/security/netexec/default.nix
@@ -8,13 +8,18 @@ let
   python = python3.override {
     packageOverrides = self: super: {
       impacket = super.impacket.overridePythonAttrs {
-        version = "0.12.0.dev1";
+        version = "0.12.0.dev1-unstable-2023-11-30";
         src = fetchFromGitHub {
           owner = "Pennyw0rth";
           repo = "impacket";
           rev = "d370e6359a410063b2c9c68f6572c3b5fb178a38";
           hash = "sha256-Jozn4lKAnLQ2I53+bx0mFY++OH5P4KyqVmrS5XJUY3E=";
         };
+        # Fix version to be compliant with Python packaging rules
+        postPatch = ''
+          substituteInPlace setup.py \
+            --replace 'version="{}.{}.{}.{}{}"' 'version="{}.{}.{}"'
+        '';
       };
     };
   };

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -3388,6 +3388,8 @@ self: super: with self; {
 
   dpkt = callPackage ../development/python-modules/dpkt { };
 
+  dploot = callPackage ../development/python-modules/dploot { };
+
   draftjs-exporter = callPackage ../development/python-modules/draftjs-exporter { };
 
   dragonfly = callPackage ../development/python-modules/dragonfly { };

